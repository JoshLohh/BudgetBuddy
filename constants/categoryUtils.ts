import { Ionicons } from "@expo/vector-icons";

export const CATEGORIES = [
  { label: 'Food', value: 'Food', icon: 'restaurant' },
  { label: 'Transportation', value: 'Transportation', icon: 'car' },
  { label: 'Accommodation', value: 'Accommodation', icon: 'home' },
  { label: 'Entertainment', value: 'Entertainment', icon: 'film' },
  { label: 'Utilities', value: 'Utilities', icon: 'bulb' },
  { label: 'Shopping', value: 'Shopping', icon: 'cart' },
  { label: 'Others', value: 'Others', icon: 'ellipsis-horizontal' },
];

export const getCategoryIconName = (category: string) => {
  const found = CATEGORIES.find(c => c.value === category);
  return found?.icon || 'ellipsis-horizontal';
};

export type IoniconName = React.ComponentProps<typeof Ionicons>['name'];