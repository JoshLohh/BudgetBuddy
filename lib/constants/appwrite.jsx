import { Client, Account, ID, Avatars } from 'react-native-appwrite';


export const client = new Client
    .setProject("6838ad0d003d2a0754a0")
    .setPlatform('dev.Buddies.budgetbuddy');

export const account = new Account(client)
export const avatars = new Avatars(client)