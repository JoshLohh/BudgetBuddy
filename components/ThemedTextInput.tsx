import { useColorScheme, TextInput, TextInputProps } from 'react-native'
import React from 'react'
import { Colors } from '../constants/Colors'

const ThemedTextInput = ({ style, ...props }: TextInputProps) => {
    const colorScheme = useColorScheme()
    const theme = Colors[colorScheme === "dark" ? "dark" : "light" ] ?? Colors.light
  
    return (
        <TextInput 
            style = {[
                {
                    backgroundColor: theme.uiBackGround,
                    color: theme.text,
                    padding: 20,
                    borderRadius: 6,
                },
                style
            ]}
            {...props}
        />
  )

}

export default ThemedTextInput

