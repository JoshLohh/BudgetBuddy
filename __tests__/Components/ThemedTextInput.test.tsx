import React from 'react';
import { render } from '@testing-library/react-native';
import * as ReactNative from 'react-native';
import { Colors } from '../../constants/Colors';
import ThemedTextInput from '../../components/ThemedTextInput';

describe('ThemedTextInput', () => {
  afterEach(() => {
    jest.restoreAllMocks();
  });

  it('applies correct styles in light mode', () => {
    jest.spyOn(ReactNative, 'useColorScheme').mockReturnValue('light');

    const { getByPlaceholderText } = render(
      <ThemedTextInput placeholder="Light input" />
    );
    const input = getByPlaceholderText('Light input');
    const styles = Array.isArray(input.props.style) ? input.props.style : [input.props.style];

    expect(styles).toEqual(
      expect.arrayContaining([
        expect.objectContaining({
          backgroundColor: Colors.light.uiBackGround,
          color: Colors.light.text,
          padding: 20,
          borderRadius: 6,
        }),
      ])
    );
  });

  it('applies correct styles in dark mode', () => {
    jest.spyOn(ReactNative, 'useColorScheme').mockReturnValue('dark');

    const { getByPlaceholderText } = render(
      <ThemedTextInput placeholder="Dark input" />
    );
    const input = getByPlaceholderText('Dark input');
    const styles = Array.isArray(input.props.style) ? input.props.style : [input.props.style];

    expect(styles).toEqual(
      expect.arrayContaining([
        expect.objectContaining({
          backgroundColor: Colors.dark.uiBackGround,
          color: Colors.dark.text,
          padding: 20,
          borderRadius: 6,
        }),
      ])
    );
  });

  it('falls back to light theme if useColorScheme returns undefined', () => {
    jest.spyOn(ReactNative, 'useColorScheme').mockReturnValue(undefined);

    const { getByPlaceholderText } = render(
      <ThemedTextInput placeholder="Fallback input" />
    );
    const input = getByPlaceholderText('Fallback input');
    const styles = Array.isArray(input.props.style) ? input.props.style : [input.props.style];

    expect(styles).toEqual(
      expect.arrayContaining([
        expect.objectContaining({
          backgroundColor: Colors.light.uiBackGround,
          color: Colors.light.text,
          padding: 20,
          borderRadius: 6,
        }),
      ])
    );
  });

  it('merges custom styles with default styles', () => {
    jest.spyOn(ReactNative, 'useColorScheme').mockReturnValue('light');

    const customStyle = { margin: 12, borderWidth: 2 };
    const { getByPlaceholderText } = render(
      <ThemedTextInput placeholder="Styled input" style={customStyle} />
    );
    const input = getByPlaceholderText('Styled input');
    const styles = Array.isArray(input.props.style) ? input.props.style : [input.props.style];

    expect(styles).toEqual(
      expect.arrayContaining([
        expect.objectContaining(customStyle),
      ])
    );
  });

  it('matches snapshot for light theme', () => {
    jest.spyOn(ReactNative, 'useColorScheme').mockReturnValue('light');

    const tree = render(<ThemedTextInput placeholder="Snapshot input" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot for dark theme', () => {
    jest.spyOn(ReactNative, 'useColorScheme').mockReturnValue('dark');
    const tree = render(<ThemedTextInput placeholder="Dark mode snapshot" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
});

/**
 * This test suite validates ThemedTextInput's behavior under different theme conditions.
 * It ensures that default and merged styles are applied correctly, theme detection works
 * via useColorScheme, and the rendered structure remains stable over time via snapshot testing.
 */
