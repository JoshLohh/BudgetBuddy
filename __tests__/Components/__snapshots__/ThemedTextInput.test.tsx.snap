// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemedTextInput matches snapshot for dark theme 1`] = `
<TextInput
  placeholder="Dark mode snapshot"
  style={
    [
      {
        "backgroundColor": "#2f2b3d",
        "borderRadius": 6,
        "color": "#ECEDEE",
        "padding": 20,
      },
      undefined,
    ]
  }
/>
`;

exports[`ThemedTextInput matches snapshot for light theme 1`] = `
<TextInput
  placeholder="Snapshot input"
  style={
    [
      {
        "backgroundColor": "#d6d5e1",
        "borderRadius": 6,
        "color": "#11181C",
        "padding": 20,
      },
      undefined,
    ]
  }
/>
`;
